### DISCUSSION TITLE
type object 'Token' has no attribute 'DoesNotExist'

### DISCUSSION THREAD

**Discussion by buzzlightyear2k** [netbox-community/netbox#10010] [Q&A]
Im trying to call API and encounter this.Anyone knows how to fix it ? 
![image](https://user-images.githubusercontent.com/86591649/184575073-9067cbe2-a1ca-4d20-926e-40a986851f02.png)

**Comment by candlerb** [comment]
You'll need to show exactly how to reproduce the error - i.e. exactly what REST API request you sent.

Also, it's very helpful if you can generate a full python backtrace.  You can do this by setting the `EMAIL` and `ADMINS` sections in `configuration.py` - then the full backtrace will be emailed automatically whenever an exception occurs.

Are you sure you're running netbox within a virtualenv, as the installation instructions show?  Attempting to run it with system packages of django etc is almost guaranteed to fail.

**Reply by markkuleinio** [reply]
What he said + there is a missing slash at the end of the URL I think

**Reply by candlerb** [reply]
True, although if that were the only problem then it would be a Netbox bug. Normally I'd expect it to return a redirect to `/api/dcim/devices/` (even if the request is unauthenticated)

**Reply by markkuleinio** [reply]
You are correct:

```
$ curl http://netbox-test.lein.io/api/dcim/devices -Ii
HTTP/1.1 301 Moved Permanently
Server: nginx/1.18.0
Date: Mon, 15 Aug 2022 08:02:53 GMT
Content-Type: text/html; charset=utf-8
Content-Length: 0
Connection: keep-alive
Location: /api/dcim/devices/
```
